hw04-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,25 @@
+package homework;
+
+import java.util.*;
+
+public class CustomerService {
+
+    // важно подобрать подходящую Map-у, посмотрите на редко используемые методы, они тут полезны
+    private final TreeMap<Customer, String> customers = new TreeMap<>(Comparator.comparingLong(Customer::getScores));
```

Comment:
Лучше типы объявлять как интерфейсы:
```suggestion
    private final NavigableMap<Customer, String> customers =
            new TreeMap<>(Comparator.comparingLong(Customer::getScores));
```